### PR TITLE
Make OCP\Http strict

### PR DESCRIPTION
**Pull request by rullzer**
* Handle private files
* Add return types
* Add scalar typehints
* Made strict
* Fixed requiring proper guzzle message interface that is passed around

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: lib/private/Http/Client/Client.php [hunk 1]
```diff
@@ -133,7 +135,7 @@ private function getProxyUri() {
 	 * @return Response
```
Comment (juliushaertl) [inline lib/private/Http/Client/Client.php]: The PHPdoc comments should also use the interface instead of the implementation.

Comment (juliushaertl) [reply lib/private/Http/Client/Client.php]: Same for the other cases :wink:

**Review by juliusknorr — DISMISSED** [review]

File: lib/private/Http/Client/Client.php (hunk 1 above)
Comment (rullzer) [reply lib/private/Http/Client/Client.php]: Actually the implementation should just be return type hinted...

**Review by ChristophWurst — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Ref (#7392)

**Pull request merged**